### DISCUSSION TITLE
RLS on select with IF

### DISCUSSION THREAD

**Discussion by EdgarsLv** [supabase/supabase#6467] [Questions] [answered]
I have users table. 
It is possible to write RLS that user can select his own row: uid() = id and if certain column has true value, then can select all other users?

**Comment by GaryAustin1** [comment, marked as answer]
If you are saying your table row is  id, stuff1,stuff2,admin
Then you have to use a security definer function to test for admin as part of your policy.
(auth.uid() = id) OR is_admin()
```
CREATE OR REPLACE FUNCTION is_admin()
        returns boolean AS
        set search_path = public
        $$
        select admin from users
        where id = auth.uid()
        $$ stable language sql security definer;
```
This link shows the basic problem and idea, but I made the function more secure by not passing in a uuid and setting search path so it can't be used badly by rpc call.
https://github.com/supabase/supabase/discussions/1138
